### DISCUSSION TITLE
Remove all values from xarray based on values of another list.

### DISCUSSION THREAD

**Discussion by sagunkayastha** [pydata/xarray#7417] [Q&A] [answered]
I have a xarray DataArray with shape(1500,1600) read from a (MODIS data) netcdf file. I have another list with QA values [0, 1,23,45..] .
 I want to change all the values from the DataArray to nan, if the values are not present in QA list. 

I can perform the task looping through the list and then removing one at a time. I was wondering if there is a better way to do this? 

Thank you.

**Comment by jhamman** [comment, marked as answer]
You can use `DataArray.isin` for this:

```python
da.where(~da.isin([0, 1,23,45, ...]))
```

**Reply by sagunkayastha** [reply]
Thank you, just what i needed.
